File: examples/javascript/stl_ex.js
// Local dev: build first (`just build-javascript`), then run this example.
const octra = require("../../index.js");

var n = 10;
var v = new octra.DVector(n);
for(let i=0; i<n; i++) {
  v.set(i, i*1.1);
}
for(let i=0; i<n; i++) {
  console.log(v.get(i));
}

var v2 = new octra.IVector(n);
for(let i=0; i<n; i++) {
  v2.set(i, i*1.5);
}
for(let i=0; i<n; i++) {
  console.log(v2.get(i));
}

var p = new octra.DPair(3.14, 2.71);
console.log(p.first);
console.log(p.second);

var p2 = new octra.IPair(42, 7);
console.log(p2.first);
console.log(p2.second);

octra.hello();

File: examples/python/stl_ex.py
from pyoctra import octra

n = 100
v = octra.DVector(n)

for i in range(n):
    v[i] = i * 1.5

for i in range(n):
    print(v[i])

v2 = octra.IVector(n)
for i in range(n):
    v2[i] = int(i * 1.5)
for i in range(n):
    print(v2[i])

p = octra.IPair(3, 4)
p
print(f"p: ({p.first}, {p.second})")

p2 = octra.DPair(10, 20)
p2
print(f"p2: ({p2.first}, {p2.second})")

